mlir/test/Dialect/AMDGPU/resolve-shaped-type-result-dims.mlir

```diff
@@ -0,0 +1,33 @@
+// RUN: mlir-opt -resolve-shaped-type-result-dims -split-input-file %s | FileCheck %s
+
+func.func @fat_raw_buffer_cast_static_dim(%arg0: memref<2x3xf32>) -> (index, index) {
+  %c0 = arith.constant 0 : index
+  %c1 = arith.constant 1 : index
+  %cast = amdgpu.fat_raw_buffer_cast %arg0 : memref<2x3xf32>
+      to memref<2x3xf32, #amdgpu.address_space<fat_raw_buffer>>
+  %d0 = memref.dim %cast, %c0 : memref<2x3xf32, #amdgpu.address_space<fat_raw_buffer>>
```

Comment:
The way this will actually end up working is that you will end up calling `reifyResultShapes` twice, each for the dim operation and you will throw away the non-reified dimensions. It is "better" to just implement `reifyDimOfResult`.